File: exams/u2/chiriboga/OOPEXAM2EXERCISE14539/ExecutableCode/OOP14539Exam2Exercise/src/main/java/ec/edu/espe/oop14539exam2exercise/controller/SoccerPlayerController.java
package ec.edu.espe.oop14539exam2exercise.controller;

import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import ec.edu.espe.oop14539exam2exercise.model.SoccerPlayer;
import org.bson.Document;

/**
 *
 * @author Kerlly Chiriboga, ODS
 */
public class SoccerPlayerController {

    private MongoCollection<Document> collection;

    public SoccerPlayerController() {
        MongoClient mongoClient = MongoClients.create("mongodb+srv://kachiriboga:chiriboga@cluster0.mstbyy5.mongodb.net/");
        MongoDatabase database = mongoClient.getDatabase("Soccer");
        collection = database.getCollection("SoccerPlayers");
   }

    public void createPlayer(SoccerPlayer soccerPlayer) {
        Document doc = new Document("id", soccerPlayer.getId())
                .append("nombre", soccerPlayer.getName())
                .append("edad", soccerPlayer.getAge())
                .append("goles", soccerPlayer.getGoals())
                .append("partidosJugados", soccerPlayer.getMatchesPlayed())
                .append("promedio de goles por partido", soccerPlayer.calculateGoalsAverage())
                .append("eficiencia ofensiva",soccerPlayer.calculateOffensiveEfficiency());

        collection.insertOne(doc);
    }

}

File: ws/chiriboga/u2/ws33/controller/SoccerPlayerController.js
const { MongoClient } = require('mongodb');

class SoccerPlayerController {
  constructor() {
    this.uri = "mongodb+srv://kachiriboga:chiriboga@cluster0.mstbyy5.mongodb.net/";
    this.client = new MongoClient(this.uri, { useNewUrlParser: true, useUnifiedTopology: true });
    this.database = this.client.db("Soccer");
    this.collection = this.database.collection("SoccerPlayers");
  }

  async connect() {
    try {
      await this.client.connect();
      console.log("Connected to MongoDB");
    } catch (e) {
      console.error(e);
    }
  }

  async createPlayer(soccerPlayer) {
    const doc = {
      id: soccerPlayer.id,
      nombre: soccerPlayer.name,
      edad: soccerPlayer.age,
      goles: soccerPlayer.goals,
      partidosJugados: soccerPlayer.matchesPlayed,
      'promedio de goles por partido': soccerPlayer.calculateGoalsAverage(),
      'eficiencia ofensiva': soccerPlayer.calculateOffensiveEfficiency()
    };

    try {
      await this.collection.insertOne(doc);
      console.log("Soccer player inserted successfully");
    } catch (e) {
      console.error(e);
    }
  }
}

module.exports = SoccerPlayerController;

